cp23/hpi-fun2/cubes/executables/generator.cpp

```diff
@@ -0,0 +1,159 @@
+    const long long base_z_coordinate = rnd.next(MAX_COORDINATE);
+
+    // Threshold for the cloud block
+    const float THRESHOLD = 0.4;
+
+    FastNoiseLite noiseGenerator;
+
+    // Configure the noise
+    noiseGenerator.SetNoiseType(FastNoiseLite::NoiseType_ValueCubic);
+    noiseGenerator.SetFrequency(0.1);
+
+    for (int x = 0; x < width; x++)
+    {
+        for (int y = 0; y < length; y++)
+        {
+            for (int z = 0; z < height; z++)
+            {
+                // Generate noise value
+                float noise = noiseGenerator.GetNoise((float)x, (float)y, (float)z);
+
+                // If noise value is above threshold, generate a cloud block
+                if (noise > THRESHOLD)
+                    coords.push_back({base_x_coordinate + x, base_y_coordinate + y, base_z_coordinate + z});
+
+                if (coords.size() == MAX_N)
+                    break;
+            }
+        }
+    }
+
+    return coords;
+};
+
+int main(int argc, char *argv[])
+{
+    registerGen(argc, argv, 1);
+    rnd.setSeed(-1143501878491944586ll);
+
+    sample(1, SAMPLE1);
+    sample(2, SAMPLE2);
+
+    testcase("empty", "empty", []
+             { println(0); });
+
+    testcase("small-x", "two cubes differ by x", []
+             {
+        println(2);
+        println(1, 1, 1);
+        println(2, 1, 1); });
+
+    testcase("small-y", "two cubes differ by y", []
+             {
+        println(2);
+        println(1, 1, 1);
+        println(1, 2, 1); });
+
+    testcase("small-z", "two cubes differ by z", []
+             {
+        println(2);
+        println(1, 1, 1);
+        println(1, 1, 2); });
+
+    testcase("diagonals", "diagonals", []
+             {
+        println(2);
+        println(1, 1, 1);
+        println(1, 2, 2); });
+
+    testcase("complex-diagonals", "complex diagonals", []
+             {
+        println(6);
+        println(2, 2, 1);
+        println(2, 2, 3);
+        println(2, 1, 2);
+        println(2, 3, 2);
+        println(1, 2, 2);
+        println(3, 2, 2); });
+
+    testcase("max_coords", "max coordinates", []
+             {
+        println(2);
+        println(MAX_COORDINATE, MAX_COORDINATE, MAX_COORDINATE);
+        println(MAX_COORDINATE - 1, MAX_COORDINATE - 1, MAX_COORDINATE - 1); });
+
+    testcase("max_n", "max n", []
+             {
+    auto coords = generate_coords(1000, 1000, 1000);
+    int n = coords.size();
+    println(n);
+    for (int i = 0; i < n; i++)
+        println(coords[i][0], coords[i][1], coords[i][2]); });
+
+    for (int j = 0; j < 10; j++)
+    {
+
+        testcase("random" + to_string(j), "random", []
+                 {
+            auto coords = generate_coords(100 + rnd.next(100), 100 + rnd.next(100), 100 + rnd.next(100));
+            int n = coords.size();
+            println(n);
+            for(int i = 0; i < n; i++)
+                println(coords[i][0], coords[i][1], coords[i][2]); });
+    }
```

Comment:
```suggestion
    for (int j = 0; j < 10; j++)
    {

        testcase("random" + to_string(j), "random", []
                 {
            auto coords = generate_coords(100 + rnd.next(100), 100 + rnd.next(100), 100 + rnd.next(100));
            int n = coords.size();
            println(n);
            for(int i = 0; i < n; i++)
                println(coords[i][0], coords[i][1], coords[i][2]); });
    }
```

11 max testcase might be a bit slow on the judge. If you want to speed up judging consider decreasing the `n` for the random testcases (but it is also fair not to do so).